### ISSUE TITLE
cryptonite-0.24 fails to build with memory-0.14.9

### ISSUE THREAD

**Issue by mithrandi** [vincenthz/hs-memory#40]
```
Crypto/Hash/Types.hs:66:22:
    No instance for (ByteArrayAccess (F.UArray Word8))
      arising from the 'deriving' clause of a data type declaration
    Possible fix:
      use a standalone 'deriving instance' declaration,
        so you can specify the instance context yourself
    When deriving the instance for (ByteArrayAccess (Digest a))
```
I'm guessing this is the fault of #39 somehow.

**Comment by mithrandi** [comment]
Oh, this only happens on ghc 7.8.x which I guess is not supported anymore? EDIT: I see the README still says it is supported; the removal from the CI matrix is what lead me to think it was unsupported.

**Comment by vincenthz** [comment]
I missed the issue for some reason, but it has been fixed since 0.14.10
